source/particle/world.cc

```diff
@@ -553,28 +558,64 @@ namespace aspect
                           ExcMessage("Periodic boundaries combined with particles currently "
                                      "only work with 90 degree opening angle in spherical shell."));
 
+              // Find out if we need to rotate the temporary properties
+              unsigned int property_index_to_rotate = numbers::invalid_unsigned_int;
+              if (Plugins::plugin_type_matches<Particle::Integrator::Euler<dim>>(*integrator) == true)
+                {
+                  // nothing to do
+                }
+              else if (Plugins::plugin_type_matches<Particle::Integrator::RK2<dim>>(*integrator) == true)
+                {
+                  // rotate original location
+                  property_index_to_rotate = get_property_manager().get_data_info().get_position_by_field_name("internal: integrator properties");
```

Comment:
This is ugly. First, you make the assumption that the RK2 method stores a tensor in this location, but this is really something only the RK2 method should know. Second, you make the assumption that there is only one field that needs to be rotated.